src/components/confirm/confirm.ts

```diff
@@ -31,27 +46,24 @@ async function nativeConfirm(options) {
     }
 }
 
-async function customConfirm(text, title) {
-    let options;
-    if (typeof text === 'string') {
+async function customConfirm(options: string | ConfirmOptions, title: string) {
+    if (typeof options === 'string') {
         options = {
             title: title,
-            text: text
+            text: options
         };
-    } else {
-        options = text;
     }
 
-    const items = [];
+    const items: OptionItem[] = [];
 
     items.push({
-        name: options.cancelText || globalize.translate('ButtonCancel'),
+        name: options.cancelText ?? globalize.translate('ButtonCancel'),
```

Comment:
One thing to be aware of is both || and ?? are now available in Javascript and neither requires a transpiler step to change ?? to || in case that's what you meant by your "(and portable)" comment?

Reply:
We need to support old web engines that don't support many modern features.
For example, Tizen 2.x and webOS 1/2 use the old WebKit (from 2015, iirc). There's no `let`, `const`, arrow functions, nullish coalescing operator, optional chaining operator and so on.
Because of that, we need to transpile the code. But in some cases it is better to use the most portable syntax - less additional code is produced.